pilot/pkg/model/authentication_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"testing"
+
+	mesh "istio.io/api/mesh/v1alpha1"
+	security_beta "istio.io/api/security/v1beta1"
+	selectorpb "istio.io/api/type/v1beta1"
+	"istio.io/istio/pkg/config/labels"
+	"istio.io/istio/pkg/config/schemas"
+)
+
+const (
+	rootNamespace = "istio-config"
+)
+
+func TestGetJwtPoliciesForWorkload(t *testing.T) {
+	policies := getTestAuthenticationPolicies(createTestConfigs(), t)
+
+	cases := []struct {
+		name              string
+		workloadNamespace string
+		workloadLabels    labels.Collection
+		want              int
+	}{
+		{
+			name:              "Empty workload labels in foo",
+			workloadNamespace: "foo",
+			workloadLabels:    labels.Collection{},
+			want:              2,
```

Comment:
shouldn't we use a map[policy-name/string]struct{} for compare the policy selection?

rather than only checking the length of the policies matched?

Reply:
done